### PR TITLE
MON-3990: make TestClusterMonitorConsolePlugin check that one of the pods can serve /plugin-manifest.json

### PR DESCRIPTION
**Pull request by machine424**
… pods can serve /plugin-manifest.json

This is all the more important since the plugin Pods do not have readiness probes.

A more realistic test would ensure the console can reach the plusgins, that can be added in origin/

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: test/e2e/config_test.go [hunk 1]
```diff
@@ -727,6 +731,40 @@ func TestUserWorkloadMonitorThanosRulerConfig(t *testing.T) {
...
+			return err
+		}
+		defer resp.Body.Close()
+
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("fail to read response body: %w", err)
+		}
+
+		if resp.StatusCode != http.StatusOK {
```
Comment (simonpasquier) [inline test/e2e/config_test.go:760]: (nit) I would check the status code first and then verify that the response body length is > 0 (or even valid JSON?).

File: test/e2e/config_test.go [hunk 2]
```diff
@@ -727,6 +732,49 @@ func TestUserWorkloadMonitorThanosRulerConfig(t *testing.T) {
...
+			return fmt.Errorf("expected status %d, got %d (%q)", resp.StatusCode, http.StatusOK, framework.ClampMax(b))
+		}
+
+		res, err := gabs.ParseJSON(b)
+		if err != nil {
+			return err
+		}
+		name, ok := res.Path("name").Data().(string)
+		if !ok || name != pluginName {
+			t.Errorf("expected plugin name to be %s, got `%s`", pluginName, name)
```
Comment (simonpasquier) [inline test/e2e/config_test.go:771]: (nit) if !ok then name is empty? Also shouldn't we return an err rather than calling t.Errorf()?

Comment (machine424) [reply test/e2e/config_test.go:771]: Good catch for the `t.Errorf`
> (nit) if !ok then name is empty

That's why I'm putting it between ``

Comment (machine424) [reply test/e2e/config_test.go]: `t.Errorf` still makes the test fail, but it should be cleaner now.

**Comment by machine424** [comment]
/retitle MON-3990: make TestClusterMonitorConsolePlugin check that one of the pods can serve /plugin-manifest.json

**Comment by machine424** [comment]
Note that a readinessProbe will as well be added for the new Go backend https://github.com/openshift/cluster-monitoring-operator/pull/2412#issuecomment-2315085438

File: test/e2e/config_test.go [hunk 3]
```diff
@@ -727,6 +732,49 @@ func TestUserWorkloadMonitorThanosRulerConfig(t *testing.T) {
...
+			return fmt.Errorf("expected status %d, got %d (%q)", resp.StatusCode, http.StatusOK, framework.ClampMax(b))
+		}
+
+		res, err := gabs.ParseJSON(b)
+		if err != nil {
+			return err
+		}
+		name, ok := res.Path("name").Data().(string)
+		if !ok || name != pluginName {
+			return fmt.Errorf("expected plugin name to be %s, got `%s`", pluginName, name)
```
Comment (simonpasquier) [inline test/e2e/config_test.go:771]: (nit++)
```suggestion
			return fmt.Errorf("expected plugin name to be %q, got %q", pluginName, name)
```

Comment (machine424) [reply test/e2e/config_test.go]: noooooo

File: test/e2e/config_test.go [hunk 4]
```diff
@@ -727,6 +732,48 @@ func TestUserWorkloadMonitorThanosRulerConfig(t *testing.T) {
...
+			return fmt.Errorf("expected status %d, got %d (%q)", resp.StatusCode, http.StatusOK, framework.ClampMax(b))
+		}
+
+		res, err := gabs.ParseJSON(b)
+		if err != nil {
+			return err
+		}
+		name, ok := res.Path("name").Data().(string)
+		if !ok || name != pluginName {
+			return fmt.Errorf("expected plugin name to be %q, got `%q`", pluginName, name)
```
[the remark this answers is not in the document]
Comment (machine424) [reply test/e2e/config_test.go]: noooooo

**Review by simonpasquier — COMMENTED** [review]
/lgtm

finally :smile:

**Pull request merged**